test/e2e/framework/syncer.go

```diff
@@ -310,42 +313,26 @@ func (sf *syncerFixture) CreateSyncTargetAndApplyToDownstream(t *testing.T) *app
 	ctx, cancelFunc := context.WithCancel(context.Background())
 	t.Cleanup(cancelFunc)
 
-	rawConfig, err := sf.upstreamServer.RawConfig()
-	require.NoError(t, err)
-
 	kcpClusterClient, err := kcpclientset.NewForConfig(syncerConfig.UpstreamConfig)
 	require.NoError(t, err)
-	var virtualWorkspaceURL string
+	var syncerVirtualWorkspaceURLs []string
+	var upsyncerVirtualWorkspaceURLs []string
 	var syncTargetClusterName logicalcluster.Name
 	Eventually(t, func() (success bool, reason string) {
 		syncTarget, err := kcpClusterClient.Cluster(syncerConfig.SyncTargetPath).WorkloadV1alpha1().SyncTargets().Get(ctx, syncerConfig.SyncTargetName, metav1.GetOptions{})
 		require.NoError(t, err)
-		if len(syncTarget.Status.VirtualWorkspaces) != 1 {
-			return false, ""
+		if len(syncTarget.Status.VirtualWorkspaces) != sf.upstreamServer.NumberOfShards() {
+			return false, fmt.Sprintf("syncTarget has %d virtual workspaces, but should have %d", len(syncTarget.Status.VirtualWorkspaces), sf.upstreamServer.NumberOfShards())
 		}
```

Comment:
is this stable under changing number of shards?

Reply:
I think yes.

Reply:
I think he's making the same comment as I am: https://github.com/kcp-dev/kcp/pull/2908#discussion_r1142469596